Example/KlarnaOnDemandExample/MainViewController.m

```diff
@@ -18,17 +18,71 @@ - (void)viewDidLoad {
   [self updateUIElements];
 }
 
+#pragma mark Button clicks
+
+- (IBAction)onBuyPressed:(id)sender {
+  // if a token has not been previously created
+  if([self hasUserToken]) {
+    [self buyTicket];
+  }
+  else {
+    [self openKlarnaRegistration];
+  }
+}
+
+- (IBAction)onChangePaymentPressed:(id)sender {
+  [self openKlarnaPreferences];
```

Comment:
This method call other method, so I think that there is no need for the other method